src/lists.c

```diff
@@ -1797,6 +1797,63 @@ void AsssListLevelCheck (
 }
 
 
+/****************************************************************************
+**
+*F  FuncSWAP_MAT_ROWS( <self>, <mat>, <row1>, <row2> )
+*/
+static Obj SwapMatRows;
+
+static Obj FuncSWAP_MAT_ROWS(Obj self, Obj mat, Obj row1, Obj row2)
+{
+    if (IS_POS_INTOBJ(row1) && IS_POS_INTOBJ(row2) && IS_PLIST(mat)) {
+        Int r1 = INT_INTOBJ(row1);
+        Int r2 = INT_INTOBJ(row2);
+        if (r1 <= LEN_PLIST(mat) && r2 <= LEN_PLIST(mat)) {
```

Comment:
This is wrong: if one is in range and the other is not, we now do nothing. But either we should swap them (possibly requiring some resizing), or report an error.